docker/docker.go

```diff
@@ -24,11 +24,118 @@ const (
 	defaultCertFile     = "cert.pem"
 )
 
+func checkDaemon(args []string) bool {
+	for j := 1; j < len(args); j++ {
+		if args[j] == "-d" || args[j] == "--daemon" {
+			return true
+		} else if args[j][:1] == "-" {
+			if index_of_equal := strings.Index(args[j], "="); index_of_equal != -1 {
+				continue //Flag with a value
+			} else {
+				isBool, exists := flag.CommandLine.FlagsMap[args[j]]
+				if exists && !isBool {
+					//Global string flag
+					j = j + 1
+					continue
+				} else if exists && isBool {
+					//Global bool flag
+					continue
+				}
+
+				isBool, exists = cmd.FlagsMap[args[j]]
+				if exists && !isBool {
+					//Daemon string flag
+					j = j + 1
+					continue
+				} else if exists && isBool {
+					//Daemon bool flag
+					continue
+				}
+			}
+		} else {
+			for _, command := range Commands {
+				if args[j] == command[0] {
+					return false
+				}
+			}
```

Comment:
Ok, let me explain by giving a concrete example: `docker rnu myimage cmd -a -b -c -d`.
Clearly this is a misspelling of the `run` command. Unfortunately, because you only check for real subcommands, this will be (incorrectly) identified as a command to shuffle into a daemon command.
My proposal is: as soon as you find any argument which isn't a daemon or global argument we should assume that this is *not* a daemon command. This correctly handles a) valid subcommands b) misspelled subcommands.